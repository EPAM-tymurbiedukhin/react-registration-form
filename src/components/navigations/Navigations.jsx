import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import classnames from 'classnames';

import {
  REGISTRATION_PAGE,
  OTHER_INFORMATION_PAGE,
  PASSWORD_QUESTIONS_PAGE,
} from '../../constants/path-constants';

import { footer, activeLink } from './navigations.module.scss';

export default () => {
  const registration = useSelector((state) => state.registration.isFormValid);
  const registrationQuestions = useSelector((state) => state.registrationQuestions.isFormValid);

  return (
    <footer className={footer}>
      <NavLink
        to={REGISTRATION_PAGE}
        activeClassName={activeLink}
        className={classnames('waves-effect waves-light btn-large')}
      >
        Registration
      </NavLink>
      <NavLink
        to={PASSWORD_QUESTIONS_PAGE}
        activeClassName={activeLink}
        className={classnames('waves-effect waves-light btn-large', !registration && 'disabled')}
      >
        Registration-questions
      </NavLink>
      <NavLink
        to={OTHER_INFORMATION_PAGE}
        activeClassName={activeLink}
        className={classnames('waves-effect waves-light btn-large', !registrationQuestions && 'disabled')}
      >
        Registration-info
      </NavLink>
    </footer>
  );
};
