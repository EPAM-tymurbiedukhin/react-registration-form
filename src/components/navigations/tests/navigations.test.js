import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import {
  REGISTRATION_PAGE,
  PASSWORD_QUESTIONS_PAGE,
  OTHER_INFORMATION_PAGE,
} from '../../../constants/path-constants';

import Navigations from '../Navigations';

import { footer } from '../navigations.module.scss';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

const setUp = () =>
  mount(
    <BrowserRouter>
      <Navigations />
    </BrowserRouter>
  );

const linksProperties = [
  { href: REGISTRATION_PAGE, text: 'Registration' },
  {
    href: PASSWORD_QUESTIONS_PAGE,
    text: 'Registration-questions',
  },
  { href: OTHER_INFORMATION_PAGE, text: 'Registration-info' },
];

describe('<MainInput />', () => {
  let component;

  beforeEach(() => {
    // useSelector.mockReturnValueOnce(mockedUseSelectorValue);
    component = setUp();
  });

  it('should render correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  it('should footer has correct className', () => {
    expect(component.find('footer').hasClass(footer)).toBeTruthy();
  });

  it('should all NavLinks has correct class', () => {
    component.find('a').forEach((node) => {
      expect(node.hasClass('waves-effect waves-light btn-large')).toBeTruthy();
    });
  });

  it('should all NavLinks render correctly', () => {
    component.find('a').forEach((node, i) => {
      expect(node.hasClass('waves-effect waves-light btn-large')).toBeTruthy();
      expect(node.props().href).toBe(linksProperties[i].href);
      expect(node.text()).toBe(linksProperties[i].text);
    });
  });

  it('should second NavLink disable', () => {
    useSelector.mockReturnValueOnce(false);
    component = setUp();
    expect(component.find('a').at(1).hasClass('disabled')).toBeTruthy();
  });

  it('should third NavLink disable', () => {
    useSelector.mockReturnValueOnce(true);
    useSelector.mockReturnValueOnce(false);
    component = setUp();
    expect(component.find('a').at(2).hasClass('disabled')).toBeTruthy();
  });
});
