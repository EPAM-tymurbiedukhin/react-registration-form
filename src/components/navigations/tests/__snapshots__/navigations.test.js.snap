// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainInput /> should render correctly 1`] = `"<footer class=\\"footer\\"><a class=\\"waves-effect waves-light btn-large\\" href=\\"/registration\\">Registration</a><a class=\\"waves-effect waves-light btn-large disabled\\" href=\\"/registration-questions\\">Registration-questions</a><a class=\\"waves-effect waves-light btn-large disabled\\" href=\\"/registration-info\\">Registration-info</a></footer>"`;
